services/src/main/scala/cromwell/services/instrumentation/AsynchronousThrottlingGaugeMetricActor.scala

```diff
@@ -0,0 +1,73 @@
+package cromwell.services.instrumentation
+
+import akka.actor.{ActorRef, LoggingFSM, Props, Status}
+import akka.pattern.pipe
+import cats.data.NonEmptyList
+import cromwell.services.instrumentation.AsynchronousThrottlingGaugeMetricActor._
+
+import scala.concurrent.Future
+
+class AsynchronousThrottlingGaugeMetricActor(metricPath: NonEmptyList[String],
+                                             instrumentationPrefix: Option[String],
+                                             override val serviceRegistryActor: ActorRef)
+  extends LoggingFSM[AsynchronousThrottlingGaugeMetricActorState, Unit]
+    with CromwellInstrumentation {
+
+  implicit val ec = context.dispatcher
+
+  startWith(WaitingForMetricCalculationRequestOrMetricValue, ())
+
+  when(WaitingForMetricCalculationRequestOrMetricValue) {
+    case Event(MetricValue(value), _) =>
+      sendGaugeAndStay(value)
+    case Event(CalculateMetricValue(calculateMetricValueFunction), _) =>
```

Comment:
Based on the two uses so far, I think it would be ok to make this class a `trait` and ask for an implementation of `calculateMetricValueFunction` in the `trait` definition.

I don't think I can see a situation where a single metrics generating actor will be shared between two different types of metric generation?

Reply:
> ask for an implementation of calculateMetricValueFunction in the trait definition.

A problem with this implement/override approach is that two currently existing implementations of `calculateMetricValueFunction` accept different parameters (one requires no parameters at all, and another requires `OffsetDateTime`, which would be different on each invocation).

Reply:
Oh, I see. TOL, I wonder whether these are really two different interfaces then...?

I'm always a little nervous about passing functions around in messages because that's one reason why it's been so hard to separate our various components into sub-units, and it makes the tracing through of "where is this method actually coming from" harder to debug when we have problems. I'm especially nervous when those functions are closures around state that's coming in from the message-sending actor.

In this case it's all fairly self-contained and only used in a few places so I think it'll probably be ok... just be aware that we've been bitten before where we try to be clever with "functions as arguments" to try to make general purpose abstraction classes, when really things would be much simpler and easier to reason about if we just had two classes which looked fairly similar (but not exactly the same)